backend/rest/rest.go

```diff
@@ -36,49 +36,61 @@ func New(token string, resultsPerPage int, storageProvider persistence.Storage)
 }
 
 // Search searches code using the github api
-func (c *client) Search(query string, lang string) []string {
+func (c *client) Search(query string, lang string) map[int64]*backends.Result {
 
 	opt := &github.SearchOptions{
 		ListOptions: github.ListOptions{PerPage: c.resultsPerPage},
 	}
 
-	var repos []string
 	results, _, err := c.Client.Search.Code(c.ctx, query, opt)
+	repos := make(map[int64]*backends.Result)
+
 	if err != nil {
 		log.Fatal("Failed to fetch results", err)
 	}
 
-	for _, result := range results.CodeResults {
-		repos = append(repos, result.GetRepository().GetHTMLURL())
-		repoID := result.GetRepository().GetID()
-		fmt.Printf("Fetching value for: %v", repoID)
+	for _, codeResult := range results.CodeResults {
```

Comment:
Can you do a multi-get for all the ids from storage?

Reply:
what do you mean?

Reply:
You're doing:
foreach (repo) { handleResult(storage.get(repo.id)); }

I suggest:
ids = repos.map(repo => repo.id).toList();
results = storage.get(ids);
handleResults(results);

Reply:
yeah, good idea